crates/sncast/src/starknet_commands/account/import.rs

```diff
@@ -58,36 +57,33 @@ pub struct Add {
     pub rpc: RpcArgs,
 }
 
-pub async fn add(
+pub async fn import(
     account: &str,
     accounts_file: &Utf8PathBuf,
     provider: &JsonRpcClient<HttpTransport>,
-    add: &Add,
-) -> Result<AccountAddResponse> {
-    let private_key = match &add.private_key_file_path {
-        Some(file_path) => get_private_key_from_file(file_path)
-            .with_context(|| format!("Failed to obtain private key from the file {file_path}"))?,
-        None => add
-            .private_key
-            .expect("Failed to parse provided private key"),
+    import: &Import,
+) -> Result<AccountImportResponse> {
+    let private_key = match (&import.private_key, &import.private_key_file_path) {
+        (Some(passed_private_key), None) => passed_private_key,
+        (None, Some(passed_private_key_file_path)) => &{
+            get_private_key_from_file(passed_private_key_file_path).with_context(|| {
+                format!("Failed to obtain private key from the file {passed_private_key_file_path}")
+            })?
+        },
+        (None, None) => &get_private_key_from_input(),
+        _ => unreachable!("Checked on clap level"),
     };
-    let private_key = &SigningKey::from_secret_scalar(private_key);
-    if let Some(public_key) = &add.public_key {
-        ensure!(
-            public_key == &private_key.verifying_key().scalar(),
-            "The private key does not match the public key"
-        );
-    }
+    let private_key = &SigningKey::from_secret_scalar(*private_key);
 
-    let fetched_class_hash = get_class_hash_by_address(provider, add.address).await?;
+    let fetched_class_hash = get_class_hash_by_address(provider, import.address).await?;
     let deployed = fetched_class_hash.is_some();
-    let class_hash = match (fetched_class_hash, add.class_hash) {
+    let class_hash = match (fetched_class_hash, import.class_hash) {
```

Comment:
The whole match below is weird as well, why not write it as

```rust
let fetched_class_hash = get_class_hash_by_address(provider, import.address).await?;

if let Some(provided_class_hash) = import.class_hash {
    ensure!(
                fetched_class_hash == provided_class_hash,
                "Incorrect class hash {:#x} for account address {:#x}",
                from_user,
                import.address
            );
            fetched_class_hash
}
```

I don't understand in which case would we not be able to fetch the class_hash for given address and at the same time be able to correctly validate the one that the user provided. I think they must match, no?

Reply:
Also it should probably say that the incorrect class hash **was provided**.

Reply:
> I don't understand in which case would we not be able to fetch the class_hash for given address and at the same time be able to correctly validate the one that the user provided. I think they must match, no?

When an account contract is declared and not deployed, we will not be able to fetch its class hash so such situation can surely happen. That's why we later check whether it's declared.

```rust
 (None, Some(from_user)) => {
            check_class_hash_exists(provider, from_user).await?;
            Some(from_user)
        }
```

Reply:
I refactored class hash resolution, it should be better now.